src/hinting.ts

```diff
@@ -157,8 +178,56 @@ class Hint {
     }
 }
 
-/** Show only hints prefixed by fstr. Focus first match */
+type HintBuilder = (els: Element[], onSelect: HintSelectedCallback) => void
+
+function buildHintsSimple(els: Element[], onSelect: HintSelectedCallback) {
+    let names = hintnames(els.length)
+    for (let [el, name] of izip(els, names)) {
+        console.log({el, name})
+        modeState.hintchars += name
+        modeState.hints.push(new Hint(el, name, null, onSelect))
+    }
+}
+
+function buildHintsVimperator(els: Element[], onSelect: HintSelectedCallback) {
+    let names = hintnames(els.length)
+    const filterableTextFilter = new RegExp('[' + config.get('hintchars') + ']|[^[:alnum:]]', 'gi')
```

Comment:
Ah, I forgot about this. If someone tries to use `]` as a hintchar this will do strange things. That suggests a feature ("filter this text out the filterable text when in vimperator-style hintfiltermodes"), but right here I should probably do something about it.

...Is there a common wisdom for escaping strings for interpolation into regex character classes? o_o

Reply:
Not really: https://stackoverflow.com/questions/3446170/escape-string-for-use-in-javascript-regex#answer-6969486

The general approach is not to use user-provided strings in a RegEx, but to use `find` and its friends instead.